### PR TITLE
Add bootstrap confidence intervals for prior estimators

### LINKED ISSUE #113: Add bootstrap CIs for π + reproducibility controls (seeded resampling)
## Context / Problem
Point estimates of `pi` can be unstable and hide uncertainty, especially on
small or noisy datasets.

## Proposed Change
Add bootstrap confidence intervals for `pi` estimators with reproducibility
controls (`random_state`, deterministic resampling pathways).

## Acceptance Criteria
- [ ] Bootstrap CI helper supports all `pulearn.priors` estimators.
- [ ] Deterministic results with fixed random seeds.
- [ ] Configurable bootstrap parameters (iterations, confidence level).
- [ ] Clear warnings for low-sample/degenerate bootstrap behavior.

## Implementation Plan
- Add shared bootstrap routine and result schema.
- Integrate with estimator outputs and convenience wrappers.
- Add seeded RNG plumbing end-to-end.
- Add stability and determinism tests.

## Files / Modules Likely Touched
- `src/pulearn/priors/` CI/bootstrap utilities
- Prior estimator result object definitions
- `tests/` bootstrap and reproducibility tests

## Tests
- Add deterministic bootstrap tests using fixed seeds.
- Add CI coverage tests with synthetic known priors.
- Add edge-case tests for degenerate resamples.

## Docs
- Document CI interpretation and caveats.
- Provide examples of sensitivity to bootstrap settings.

## Backwards Compatibility / Deprecations
Additive capability; no deprecations expected.

## Notes / References
- Depends on: #112.
- Milestone: 2 — Priors & Propensity as First-Class.

### PR DESCRIPTION
**Pull request by shaypal5**
## Summary
- add a reusable stratified bootstrap CI module for `pulearn.priors`
- expose `BasePriorEstimator.bootstrap(...)` and persist CI metadata on `PriorEstimateResult`
- document the new workflow and cover deterministic seeded resampling, warnings, and failure paths

## Testing
- `ruff check src/pulearn/__init__.py src/pulearn/priors tests/test_priors.py`
- `PYTHONPATH=src JOBLIB_MULTIPROCESSING=0 python -m pytest tests/test_priors.py`
- `PYTHONPATH=src JOBLIB_MULTIPROCESSING=0 python -m pytest`

Closes #113

### CODE REVIEW

File: src/pulearn/priors/bootstrap.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    alpha = 0.5 * (1.0 - confidence_level)
+    lower, upper = np.quantile(estimates, [alpha, 1.0 - alpha])
+    return PriorConfidenceInterval(
+        lower=float(lower),
+        upper=float(upper),
+        confidence_level=float(confidence_level),
+        n_resamples=int(n_resamples),
+        successful_resamples=int(estimates.shape[0]),
+        random_state=None if random_state is None else int(random_state),
```
[the remark this answers is not in the document]
Comment (shaypal5) [reply src/pulearn/priors/bootstrap.py]: Implemented in commit `c9a33e1`. `bootstrap_confidence_interval(...)` now serializes `random_state` through `_serialize_random_state(...)`, preserving integer seeds and storing `None` for non-integer accepted seed types such as `np.random.RandomState`. Added regression coverage in `test_bootstrap_serializes_non_integer_random_state_as_none` and `test_serialize_random_state_helper`.

File: src/pulearn/priors/base.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            "method": self.method,
+            "n_samples": self.n_samples,
+            "n_labeled_positive": self.n_labeled_positive,
+            "positive_label_rate": self.positive_label_rate,
+            "metadata": dict(self.metadata),
+            "confidence_interval": (
+                None
+                if self.confidence_interval is None
+                else self.confidence_interval.as_dict()
+            ),
```
[the remark this answers is not in the document]
Comment (shaypal5) [reply src/pulearn/priors/base.py]: I did not expand the scope of this PR for the typing-only cleanup, but I turned it into a follow-up issue: #162. That keeps the runtime fixes in this PR focused while tracking the annotation tightening explicitly.
